core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -39,6 +40,47 @@
 @ThreadSafe
 public final class UnderFileSystemManager {
   /**
+   * An object which generates input streams to under file system files given a position. This
+   * class does not manage the life cycles of the generated streams and it is up to the caller to
+   * close the streams when appropriate.
+   */
+  private final class UnderFileSystemInputStream {
```

Comment:
my 2 cents: I think the name is quite misleading. From the methods it provides, I don't think this is really an `InputStream`, but more like an `InputStreamFactory` or `InputStreamAgent`. 
Similar comment applies to `UnderFileSystemOutputStream`